File: YourQuoteBoard.Client/src/models/rating/SpecificRating.ts
import { BookRatingCategory } from "../../enums/BookRatingCategory";
import { QuoteRatingCategory } from "../../enums/QuoteRatingCategory";

export interface SpecificRating{
    specificRatingId?: number;
    rating?: number;
    ratingCategory: QuoteRatingCategory | BookRatingCategory;
}

File: YourQuoteBoard/YourQuoteBoard/Entity/SpecificRating.cs
using YourQuoteBoard.Enums;

namespace YourQuoteBoard.Entity
{
    public class SpecificRating
    {
        public Guid SpecificRatingId { get; set; }
        public RatingCategory RatingCategory { get; set; }
        public double? Rating { get; set; }
        public Guid QuoteRatingId { get; set; }
        public QuoteRating QuoteRating { get; set; }
    }
}
